### PR TITLE
GUI selection of Region of Interest from grd, launcher with config option and example

### PR DESCRIPTION
**Pull request by steo85it**
Here is a drafty example of my processing tools, including a configuration file, launcher and GUI selection for a `.grd` input, responding to issue #25 . It should be sufficient to run `examples/launch_plapp.py` to have it run (I set a very low resolution for the map, for testing purposes, so results are not that fancy. You can set a lower one in `examples/launch_plapp.py` using the `FluxOpt.set("resolution", X km)` keyword.
Also, there are some options hardcoded in the `plapp/grd2obj.py` file that we might want to discuss/change/convert to options. Anyway, give it a try and let me know if it runs, makes sense and what would be worth changing (and merging with what Norbert set up as `grd2FF` example - I didn't put myself to it to "quickly" provide a first draft).

### CODE REVIEW

File: flux/shape.py [hunk 1]
```diff
@@ -168,7 +168,7 @@ def check_vis_1_to_N(self, i, J, eps=None):
             rayhit.prim_id == J
         )
 
-    def get_direct_irradiance(self, F0, Dsun, eps=None):
+    def get_direct_irradiance(self, F0, Dsun, basemesh=None, eps=None):
```
Comment (sampotter) [inline flux/shape.py:171]: Is this what Erwan was asking about (having another mesh to do check visibility only against)? If so, thanks for adding it! And you could you add a line to the docstring for this function explaining it?

Comment (sampotter) [reply flux/shape.py:171]: Or is this just a way to call `get_direct_irradiance` on a different 3D mesh, other than one held by the current instance of `TrimeshShapeModel`?

File: plapp/config.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# Options configuration for flux applications
+# TODO add setters and getters
+
+class FluxOpt:
+
+    __conf = {
```
Comment (sampotter) [inline plapp/config.py:6]: The convention for a private variable is `_conf` (just one underscore)

File: plapp/grd2obj.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+import pygmt as pygmt
+from matplotlib.path import Path
+
+import xarray as xr
+import netCDF4
+import dmsh
+import meshio
+
+from src.astro_trans import sph2cart
+from src.unproject_coord import unproject_stereographic
```
Comment (sampotter) [inline plapp/grd2obj.py:15]: What are these? Your local files or another project? They seem to be missing...

**Review by sampotter — COMMENTED** [review]
Thanks for cleaning up the commit.

There are a couple issues... looks like you made some changes to `get_direct_irradiance`, some module called `src` seems to be missing, and then the GUI stuff is split off into some separate directory named `plapp`? That's fine since this is a rough draft and can be cleaned up, but I'm curious to know what the idea is.

File: flux/shape.py (hunk 1 above)
Comment (steo85it) [reply flux/shape.py:171]: `having another mesh to do check visibility only against`
precisely so, it's a second, larger mesh (as it is implemented now, the full input map) which is just used to check visibility. Sure I can expand the docstring, I forgot about it.

File: plapp/grd2obj.py (hunk 3 above)
Comment (steo85it) [reply plapp/grd2obj.py:15]: ops, yes, that's a couple of 'transformations libraries' I copied from another project. Should I include them? Or rather directly include those 2 functions in the file? Either works for me...

**Comment by steo85it** [comment]
Hi Sam, sorry, rough days this week and just got to this...
`the GUI stuff is split off into some separate directory named plapp`: yes, I kept it separate since it's nor an example nor something (exclusively) related to the flux routines (it uses them, but it's not part of the bulk implementation). I guess while in doubt, I kept it separated: you can tell me where you'd prefer them to go (and possibly some of them should be split in multiple files to avoid getting messy, etc...).

**Pull request closed without merging**